infer/configuration.go

```diff
@@ -93,17 +93,23 @@ func (c *config[T]) checkConfig(ctx context.Context, req p.CheckRequest) (p.Chec
 		if req.Urn != "" {
 			req.Urn.Name()
 		}
+		defaultCheckEncoder := new(defaultCheckEncoderValue)
+		ctx = context.WithValue(ctx, defaultCheckEncoderKey{}, defaultCheckEncoder)
 		i, failures, err := t.Check(ctx, name, req.Olds, req.News)
 		if err != nil {
 			return p.CheckResponse{}, err
 		}
 
+		if defaultCheckEncoder.enc != nil {
+			encoder = *defaultCheckEncoder.enc
+		}
+
```

Comment:
I could use some comments here to understand what's going on. My best guess is:
- DefaultCheck uses the context to return an encoder that obscures secrets along with any errors it returns
- We use this encoder instead of the more general one created earlier so we don't have to apply secrets in here directly.

Reply:
I extracted this out into a function and added some comments.